include/cpp2util.h

```diff
@@ -536,21 +536,23 @@ class out {
     } \
 }(PARAM1)
 
-#define CPP2_UFCS_TEMPLATE(FUNCNAME,PARAM1,...) \
+#define CPP2_UFCS_REMPARENS(...) __VA_ARGS__
+
+#define CPP2_UFCS_TEMPLATE(FUNCNAME,TEMPARGS,PARAM1,...) \
 [](auto&& obj, auto&& ...params) { \
-    if constexpr (requires{ std::forward<decltype(obj)>(obj).template FUNCNAME(std::forward<decltype(params)>(params)...); }) { \
-        return std::forward<decltype(obj)>(obj).template FUNCNAME(std::forward<decltype(params)>(params)...); \
+    if constexpr (requires{ std::forward<decltype(obj)>(obj).template FUNCNAME CPP2_UFCS_REMPARENS TEMPARGS (std::forward<decltype(params)>(params)...); }) { \
+        return std::forward<decltype(obj)>(obj).template FUNCNAME CPP2_UFCS_REMPARENS TEMPARGS (std::forward<decltype(params)>(params)...); \
```

Comment:
I have learned that here: https://www.scs.stanford.edu/~dm/blog/va-opt.html while working on https://github.com/hsutter/cppfront/pull/47

I am checking the link you have sent and will check the proposed name.

Reply:
I don't like the `CPP2_VA_ARGS_IF_PARENS` name. `COMMA_IF_PARENS` is used when there are parentheses, then it will expand to the comma. In contrast, `CPP2_VA_ARGS_IF_PARENS` suggests that it will expand to the `__VA_ARGS__`, which is misleading. In the context of our macro, we use `__VA_ARGS__` but from the perspective of the caller it is just removing parentheses or there is also used the name `EXPAND` that might replace `REMPARENS`.

From my perspective, `REMPARENS` is more descriptive.